### PR TITLE
feature(cache): allow admin to attempt an automatic symlink to cache

### LINKED ISSUE #8638: Nudge admins toward symlinking /cache to views_simplecache

### LINKED ISSUE #8639: Provide easy way to verify cache/ symlink is working
e.g. write a file into cache dir that can only be served via symlink

### PR DESCRIPTION
**Pull request by hypeJunction**
This adds an option to Advanced settings to attempt symlinking cache to
views_simplecache. The action makes an HTTP request to a simplecache view
and validates the 200 response. If response was different, removes the symlink
and advises to refer to documentation for manual symlinking.

Fixes #8639
Fixes #8638

### CODE REVIEW

File: actions/admin/site/update_advanced.php [hunk 1]
```diff
@@ -45,6 +45,19 @@
 	elgg_disable_simplecache();
 }
 
+$cache_symlinked = is_dir(elgg_get_root_path() . 'cache/');
+if ('on' === get_input('cache_symlink_enabled') && !$cache_symlinked) {
+	$cache_symlinked = symlink(elgg_get_data_path() . 'views_simplecache/', elgg_get_root_path() . 'cache/');
```
Comment (mrclay) [inline actions/admin/site/update_advanced.php]: Use `elgg_get_cache_path()`. Better yet, we should make SimpleCache::getPath if it doesn't already exist.

Comment (hypeJunction) [inline actions/admin/site/update_advanced.php]: Done

File: actions/admin/site/update_advanced.php [hunk 2]
```diff
@@ -45,6 +45,19 @@
 	elgg_disable_simplecache();
 }
 
+$cache_symlinked = is_dir(elgg_get_root_path() . 'cache/');
+if ('on' === get_input('cache_symlink_enabled') && !$cache_symlinked) {
+	$cache_symlinked = symlink(elgg_get_cache_path(), elgg_get_root_path() . 'cache/');
```
Comment (mrclay) [inline actions/admin/site/update_advanced.php]: You do still need the views_simplecache path.

Comment (hypeJunction) [inline actions/admin/site/update_advanced.php]: Oh. Ok.

Comment (mrclay) [inline actions/admin/site/update_advanced.php]: Hence, SimpleCache::getPath would be handy :)

Comment (hypeJunction) [inline actions/admin/site/update_advanced.php]: Not in the mood for any more new API today :tired_face:

**Comment by mrclay** [comment]
Cool, I'll test this

File: languages/en.php [hunk 3]
```diff
@@ -1109,6 +1109,11 @@
 	'installation:simplecache:description' => "The simple cache increases performance by caching static content including some CSS and JavaScript files.",
 	'installation:simplecache:label' => "Use simple cache (recommended)",
 
+	'installation:cache_symlink:description' => "The symobolink link to the simple cache directory allows the server to serve static views bypassing the engine, which considerably improves performance and reduces the server load",
```
Comment (mrclay) [inline languages/en.php]: Typo "symobolink"

Comment (hypeJunction) [inline languages/en.php]: I am telling you I am becoming dyslexic. I don't these typos anymore...

**Comment by mrclay** [comment]
If I don't check "Use simple cache", but do check "Use symbolic link", nothing happens and none is checked. Ideally the option would be disabled like the Compress options.

**Comment by mrclay** [comment]
If creating the link fails, the user gets no error message.

**Comment by mrclay** [comment]
If the cache directory exists, is there a reason we shouldn't do the headers test whenever the advanced settings page is shown?

**Comment by hypeJunction** [comment]
We can. Felt like an inappropriate thing to do in a view. But we do many
inappropriate things in views so who cares.

**Comment by mrclay** [comment]
`admin/settings/advanced` might be the less inappropriate place.

**Comment by mrclay** [comment]
LGTM

**Comment by hypeJunction** [comment]
@jdalsem could you try this on one of your installs?

**Comment by hypeJunction** [comment]
It's been over two weeks. Merging.

**Pull request merged**